components/org.wso2.carbon.identity.client.attestation.filter/src/main/java/org/wso2/carbon/identity/client/attestation/filter/ClientAttestationProxy.java

```diff
@@ -110,18 +111,23 @@ public void handleMessage(Message message) {
                 try {
                     ServiceProvider serviceProvider =  getServiceProvider(clientId,
                             IdentityTenantUtil.resolveTenantDomain());
+                    ClientAttestationContext clientAttestationContext;
+                    // Attestation validation should be performed only if API-based authentication is enabled.
                     if (serviceProvider.isAPIBasedAuthenticationEnabled()) {
                         // Validate the attestation header and obtain client attestation context
-                        ClientAttestationContext clientAttestationContext = ClientAttestationServiceHolder.getInstance()
+                        clientAttestationContext = ClientAttestationServiceHolder.getInstance()
                                 .getClientAttestationService().validateAttestation(attestationHeader,
                                         serviceProvider.getApplicationResourceId(),
                                         IdentityTenantUtil.resolveTenantDomain());
-                        // Set the client attestation context in the HTTP request
-                        setContextToRequest(request, clientAttestationContext);
                     } else {
-                        throw new WebApplicationException(buildResponse("Client is not registered for " +
-                                        "API Based Authentication Service", Response.Status.BAD_REQUEST));
+                        /* In order for client attestation to be enabled it requires API-based authentication to be
+                         enabled. Therefore, if API-based authentication is disabled, client attestation is disabled.*/
+                        clientAttestationContext = new ClientAttestationContext();
```

Comment:
With this change, Response mode : direct can be called without enabling API Based Authentication is this expected behaviour?

Reply:
This check will be done in api based auth service level when UI support is available.